packages/blade/src/components/Typography/BaseText/BaseText.tsx

```diff
@@ -0,0 +1,45 @@
+import type { ReactElement } from 'react';
+import getIn from '../../../utils/getIn';
+import { useTheme } from '../../BladeProvider';
+import isEmpty from '../../../utils/isEmpty';
+import StyledBaseText from './StyledBaseText';
+import type { BaseTextProps } from './BaseText.d';
+
+const BaseText = ({
+  color,
+  fontFamily,
+  fontSize,
+  fontWeight,
+  fontStyle,
+  textDecorationLine,
+  lineHeight,
+  as,
+  children,
+  textAlign,
+}: BaseTextProps): ReactElement => {
+  const { theme } = useTheme();
+  const textColor = getIn(theme.colors, color);
+
+  if (isEmpty(textColor)) {
+    throw new Error(`[Blade:BaseText]: Invalid value ${color} for color prop passed`);
+  }
+
+  return (
+    <StyledBaseText
+      theme={theme}
+      color={textColor}
```

Comment:
Okay so this is wrong, 

the textColor is a plain string of the color value `hsla(...)` but `StyledBaseText` is set to only accept the tokenized color strings (action.primary.brand)

`StyledBaseText` & `getBaseTextStyles` 's color prop should only be a plain string. 

This is not caught because `getIn` is returning `any`

Reply:
`any` is a top type, can be assigned to anything. thus didn't throw error.

Reply:
😮 good catch. but i'm wondering the same how come TS didn't catch this error 🤔

Reply:
*Resolved in meeting*